kickstarter/android-oss
# NT-2093: UX – Empty state showing up when loading comments

# 📲 What

Empty state was showing upon entering the comments screen, and briefly when we pull to refresh. 

# 🛠 How

We only take the latest comments list after the `isFetching` returns false, which is the signal that the list has finished loading from graph. 

# 👀 See

https://user-images.githubusercontent.com/19390326/127355449-9b98ca26-cf07-441b-809a-b5c196c387df.mp4

# 📋 QA

- Pick a project with comments
- Go to comments screen
- Should see a loading indicator but should not display "No comments"
- Pull to refresh should also not show "No comments"

# Story 📖

[NNT-2093: UX – Empty state showing up when loading comments](https://kickstarter.atlassian.net/browse/NT-2093)
